api/src/test/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessorTest.java

```diff
@@ -75,6 +76,30 @@ void shouldCallDummyProcessors() {
         verify(messageRepository).findAllMessagesByTypeOrderedByDate(MAKE_PAYMENT);
         verify(messageRepository).findAllMessagesByTypeOrderedByDate(SEND_FIRST_EMAIL);
         verify(messageRepository).findAllMessagesByTypeOrderedByDate(DETERMINE_ENTITLEMENT);
+        verify(messageRepository).delete(cardMessage);
+    }
+
+    @Test
+    void shouldContinueToProcessMessagesAfterAnExceptionThrown() {
+        Message cardMessage1 = MessageTestDataFactory.aValidMessage();
+        Message cardMessage2 = MessageTestDataFactory.aValidMessage();
+        Message cardMessage3 = MessageTestDataFactory.aValidMessage();
+        given(messageRepository.findAllMessagesByTypeOrderedByDate(any()))
+                .willReturn(List.of(cardMessage1, cardMessage2, cardMessage3))
+                .willReturn(emptyList());
+        given(createNewCardDummyMessageTypeProcessor.processMessage(any()))
+                .willThrow(new RuntimeException("foo"))
+                .willReturn(MessageStatus.ERROR)
+                .willReturn(MessageStatus.COMPLETED);
+
+        //When
+        messageProcessor.processAllMessages();
+
+        //Then
+        verify(createNewCardDummyMessageTypeProcessor).processMessage(cardMessage1);
+        verify(createNewCardDummyMessageTypeProcessor).processMessage(cardMessage2);
+        verify(createNewCardDummyMessageTypeProcessor).processMessage(cardMessage3);
+        verify(messageRepository).delete(cardMessage3);
         verifyNoMoreInteractions(messageRepository);
```

Comment:
I'd also add `messageRepository` to `verifyNoMoreInteractions` as its being called multiple times and we want to make sure its called that exact number.